### ISSUE TITLE
Skill Bug

### ISSUE THREAD

**Issue by brerpenguin** [Heishe/idleraiders-second-run#734]
My tank will not use both challenging cry and guardian shield properly anymore. He has enough rage and speed for them both to run constantly, which they did until recently. It started out intermittently and then progressively got worse. I was originally able to fix it by unequipping and reequipping the skill. It has reached the point where doing this for one turns the other off immediately. I have reset the priorities, restarted the game, restarted the browser, removed other skills but these two, cleared my cache, soft reset, went to 90 WAIR and reenchented to -100% CD, nothing's changed. I can't think of anything else to try.

P.S.: Clearing the cache made my saves unuseable, even though I rechecked into Kong first.

**Comment by brerpenguin** [comment]
I forgot to mention that as far as I could tell he wasn't using saving throw or command while either of these were active either.

**Comment by hamburml** [comment]
Hello brerpenguin,

sorry to hear that :/ I think @Talisca is the better person to solve the skill problem.

You mention that your saves aren't working after you cleared your browser cache. What browser do you use? What exactly isn't working? Are the savegames, after you logged in to kong, not shown? Are they shown but can't be loaded? Or are the savegames older than you remember? What is your kongregate username? I would like to check the savegames.

Thanks!
Michael

**Comment by brerpenguin** [comment]
How do I contact @Talisca?

Cache bug: Chrome browser; it tells me it's unloadable because it was from another player or other version; the cloud game is there, but the exported saves are unusable; Kong name is brerpenguin

**Comment by hamburml** [comment]
He get's a mail from github because he was mentioned in this issue. You don't need to do anything. I also contacted him.

Can you send me a mail to michael.hamburger@pitforest.de with your exported saves? It's normal that a savegame who was created from Account A can't be loaded from Account B. I would like to check what account identification is stored internally on the exported savegames. Maybe I can fix that for you and also find out what happend.

Thanks!

**Comment by Talisca** [comment]
The only thing I can think of, are the combat system mechanics. 

Abilities of a creature will never start/run at the same time. They run in sequential order. The system will look for an executable ability at the next tick only if the creature is not already executing another ability.

Because of the low cooldown its likely that the ability with the highest priority will be executed again and again without giving the other abilities the chance to be executed. So the best course of action would be to only use abilities with a cooldown > 0 at the moment.

**Comment by brerpenguin** [comment]
I sent the save file (hopefully).

The only way I could get all four skills to go was using skills without any enchantments at all. Even a 
-cost enchantment made only that one skill operate.

**Comment by hamburml** [comment]
Yep, got it! Thanks!

I was able to load your savegame and it behaves like it should, I was able to find your account in our database and the stored id inside your savegame is the correct one.

Are you sure that you are correctly logged in into kongregate and when you load Idle Raiders: Second Run that your nickname is shown at the bottom right?

**Comment by brerpenguin** [comment]
I never noticed the bottom right thing. It showed my name in the upper right, which I guess is Kong? One thing I did think of is that I don't refresh Kong/IR2 because of this glitch. I just let it run til it autosaves. Could that have an effect?

**Comment by hamburml** [comment]
Yeah, that's the name of your current account and in your case this is the kongregate username.

You mentioned before that you relogged into Kongregate before you tried to import the save. That means that you had another window with Kong/IR2 open, is that correct? That could be in fact the problem.

Just to summarize (because I think I didn't fully understand everything):

Currently you are not playing IR2.
You want to import the savegame, but it's not working. But you see your kongregate username at the top right (which is kongregate itself) and the bottom right (which is our game knowing your user).

Thanks
Michael

**Comment by brerpenguin** [comment]
No, I only had the original window open. I just left it running while I cleared the cache. I l wanted to undo something that I didn't like. I tried to use the savegame which I had just updated before clearing cache and got the message that I couldn't use it.

**Comment by hamburml** [comment]
Oh that's what you did!

The game needs the localStorage, kongregate only recognizes you because you can identify through a cookie and our servers also check another cookie. With the clearing of your cache while you played it (when you are not playing it you can clear your cache, just remember you delete local savegames) you removed these files and our server doesn't know that "you" are "you".

You mentioned that you relogged to kongregate. How did you do that? The original window (before you cleared the cache) should still show your account name at the top right (kongregate). Therefore you did that in a new window, right? I just want to recreate the complete process, sorry for asking so many questions

You have an exported savegame on your computer which you want to load, right? You need to reload the kongregate page where IR2 inside, you can't use the old window. It needs to do the authentication procedure to get valid access. If the savegame you want to import is still not working, send the exact savegame you tried to import to my email address. I will fix that for you but I think you won't have a problem importing the savegame when you are correctly logged in to kongregate and to our servers (which is done automatically when you reload the kongregate page where IR2 is embedded).

Please don't clear your cache when the game is currently open, you can't redo some loot you found or some items you got out of a chest. Only clear your cache after patches (for example) and than reload everything anew. You can of course export your savegame before you clear your cache, just to be on the safe side.

Thanks
Michael

**Comment by brerpenguin** [comment]
Sorry about spacing about this for so long. I think you covered everything, but I don't open more than one Kong tab because it resets the chat. I replied about this, but don't see it now. I did export right before clearing cache. That export which was minutes old is the one it wouldn't let me use.

**Comment by hamburml** [comment]
Thanks for answering. And after you cleared the cache you didn't reload the Rongregate site with IR2 embedded?

The exported savegame you can't use anymore - was it the one you send me?

Does the importing work in the meantime? I saw that you are in the statistics today with 'Max Wair Level' -> does that mean it works or did you load another savegame?

**Comment by brerpenguin** [comment]
No, that was the new one I made right after the first didn't work, to make sure I had a useable one. If this happens again I will send you a copy of the one that doesn't work.

**Comment by hamburml** [comment]
Oh, so you started a new game?

Sorry for the inconvenience :/

**Comment by brerpenguin** [comment]
No, I just made a new exported save. Game's going fine still. I replaced the older with a newer version so I would have a backup the game would accept.
